core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/AbstractCapellaSearchEntry.java

```diff
@@ -0,0 +1,189 @@
+  protected AbstractCapellaSearchEntry parent;
+
+  /** true if a real match (eg. not an intermediary structural object for ordering purpose only), false otherwise */
+  protected boolean matchedOnce;
+
+  /**
+   * all AbstractCapellaSearchResultEntry children, potentially matches themselves or intermediary structural object(s)
+   * for ordering purposes only)
+   */
+  private Collection<Object> children;
+  
+  private String text;
+  
+  private IProject project;
+
+  public AbstractCapellaSearchEntry(AbstractCapellaSearchEntry _parent, Object _source, boolean _matched, IProject _project) {
+    super(_source, -1, -1);
+    parent = _parent;
+    children = new HashSet<Object>();
+    project = _project;
+    setMatchedOnce(_matched);
+  }
+
+  public AbstractCapellaSearchEntry(AbstractCapellaSearchEntry _parent, Object _source,
+      String _valuation, boolean _matched, IProject _project) {
+    this(_parent, _source, _matched, _project);
+    text = _valuation;
+  }
+  
+  public String getText() {
+    return text;
+  }
+  
+  public void setText(String _text) {
+    text = _text;
+  }
+  
+  public Collection<Object> getChildren() {
+    return children;
+  }
+
+  public IProject getProject() {
+    return project;
+  }
+  
+  /**
+   * Adds AbstractCapellaSearchResultEntry child, potentially match itself or intermediary structural object(s) for
+   * ordering purposes only.
+   */
+  public void addChildren(AbstractCapellaSearchEntry e) {
+    children.add(e);
+  }
+
+  /**
+   * Remove AbstractCapellaSearchResultEntry child, potentially match itself or intermediary structural object(s) for
+   * ordering purposes only.
+   */
+  public void removeChildren(AbstractCapellaSearchEntry e) {
+    children.remove(e);
+  }
+
+  /////////////////////////
+  // Getters / Setters
+  /////////////////////////
+
+  /**
+   * @return AbstractCapellaSearchResultEntry parent, potentially match itself or intermediary structural object(s) for
+   *         ordering purposes only.
+   */
+  public AbstractCapellaSearchEntry getParent() {
+    return parent;
+  }
+
+  /**
+   * Sets AbstractCapellaSearchResultEntry parent, potentially match itself or intermediary structural object(s) for
+   * ordering purposes only.
+   */
+  public void setParent(AbstractCapellaSearchEntry p) {
+    parent = p;
+  }
+
+  /**
+   * @return true if a real match (eg. not an intermediary structural object for ordering purpose only), false otherwise
+   */
+  public boolean wasMatchedAtleastOnce() {
+    return matchedOnce;
+  }
+
+  /**
+   * Set to true if a real match (eg. not an intermediary structural object for ordering purpose only), false otherwise
+   */
+  public void setMatchedOnce(boolean matchedOnce) {
+    this.matchedOnce = matchedOnce;
+  }
+
+  /**
+   * @return AbstractCapellaSearchResultEntry hierarchy in a root to leaf order (eg. tree like), void Stack otherwise.
+   */
+  public Stack<AbstractCapellaSearchEntry> getHierarchyFromRootToLeaf() {

```

Comment:
Deque, the unsynchronized version of Stack, should be used to be more performance efficient.

Reply:
OK.